src/SprykerEco/Zed/Braintree/Persistence/BraintreeEntityManager.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+/**
+ * MIT License
+ * For full license information, please view the LICENSE file that was distributed with this source code.
+ */
+
+namespace SprykerEco\Zed\Braintree\Persistence;
+
+use Orm\Zed\Braintree\Persistence\SpyPaymentBraintreeTransactionOrderItem;
+use Spryker\Zed\Kernel\Persistence\AbstractEntityManager;
+
+/**
+ * @method \SprykerEco\Zed\Braintree\Persistence\BraintreePersistenceFactory getFactory()
+ */
+class BraintreeEntityManager extends AbstractEntityManager implements BraintreeEntityManagerInterface
+{
+    /**
+     * @param int $idPaymentBraintree
+     * @param bool $isShipmentPaid
+     *
+     * @return void
+     */
+    public function updateIsShipmentPaidValue(int $idPaymentBraintree, bool $isShipmentPaid): void
+    {
+        $paymentBraintreeEntity = $this->getFactory()->createPaymentBraintreeQuery()->findOneByIdPaymentBraintree($idPaymentBraintree);
+
+        if ($paymentBraintreeEntity) {
+            $paymentBraintreeEntity->setIsShipmentPaid($isShipmentPaid);
+            $paymentBraintreeEntity->save();
+        }
+    }
+
+    /**
+     * @param int $idPaymentBraintree
+     * @param \Generated\Shared\Transfer\ItemTransfer[] $itemTransfers
+     * @param string $transactionId
+     *
+     * @return void
+     */
+    public function addOrderItemsToTransactionLog(int $idPaymentBraintree, iterable $itemTransfers, string $transactionId): void
+    {
+        $paymentBraintreeTransactionStatusLogEntity = $this->getFactory()
+            ->createPaymentBraintreeTransactionStatusLogQuery()
+            ->filterByTransactionId($transactionId)
+            ->findOneByFkPaymentBraintree($idPaymentBraintree);
+
+        if ($paymentBraintreeTransactionStatusLogEntity) {
```

Comment:
This is can be optimized for sure. Fetching data from DB in loop can lead to performance issues.

Reply:
Do you have an idea how it can be changed?